tasks/eth/base-002-fp-upgrade/NestedSignFromJson.s.sol

```diff
@@ -0,0 +1,296 @@
+        try vm.readFile(string.concat(vm.projectRoot(), path)) returns (string memory data) {
+            chainConfig = data;
+        } catch {
+            revert(string.concat("Failed to read ", path));
+        }
+
+        _proxies.ProtocolVersions = stdToml.readAddress(chainConfig, "$.protocol_versions_addr");
+        _proxies.SuperchainConfig = stdToml.readAddress(chainConfig, "$.superchain_config_addr");
+    }
+
+    function getAllowedStorageAccess() internal view override returns (address[] memory allowed) {
+        allowed = new address[](6);
+        allowed[0] = address(proxies.OptimismPortal2);
+        allowed[1] = address(proxies.SystemConfig);
+        allowed[2] = address(ownerSafe);
+        allowed[3] = address(securityCouncilSafe);
+        allowed[4] = address(fndSafe);
+        // allowed[4] = expectedLivenessGuard;
+    }
+
+    function _postCheck(Vm.AccountAccess[] memory accesses, SimulationPayload memory) internal view override {
+        console.log("Running post-deploy assertions");
+        checkSemvers();
+        checkStateDiff(accesses);
+        checkSystemConfig();
+        checkOptimismPortal();
+        console.log("All assertions passed!");
+    }
+
+    function checkStateDiff(Vm.AccountAccess[] memory accountAccesses) internal view override {
+        require(accountAccesses.length > 0, "No account accesses");
+
+        super.checkStateDiff(accountAccesses);
+
+        // Assert that only the SystemConfig, OptimismPortal and various safes storage are written to
+        for (uint256 i; i < accountAccesses.length; i++) {
+            Vm.AccountAccess memory accountAccess = accountAccesses[i];
+            require(
+                accountAccess.oldBalance == accountAccess.newBalance,
+                string.concat("Unexpected balance change: ", vm.toString(accountAccess.account))
+            );
+            require(
+                accountAccess.kind != VmSafe.AccountAccessKind.SelfDestruct,
+                string.concat("Self-destructed account: ", vm.toString(accountAccess.account))
+            );
+
+            for (uint256 j; j < accountAccess.storageAccesses.length; j++) {
+                Vm.StorageAccess memory storageAccess = accountAccess.storageAccesses[j];
+                if (storageAccess.isWrite) {
+                    address account = storageAccess.account;
+                    require(
+                        account == _ownerSafe() || account == councilSafe || account == foundationSafe
+                            || account == proxies.SystemConfig || account == proxies.OptimismPortal,
+                        "state-000"
+                    );
+                }
+            }
+        }
+    }
+
+    function checkSemvers() internal view {
+        // These are the expected semvers based on the `op-contracts/v1.4.0-rc.4` release.
+        // https://github.com/ethereum-optimism/optimism/releases/tag/op-contracts%2Fv1.4.0-rc.4
```

Comment:
These semvers should be based on the 1.6.0 tag, need to check if the comment is out of date, or the assertions: https://github.com/ethereum-optimism/optimism/releases/tag/op-contracts%2fv1.6.0

Reply:
it was the comment that was incorrect, I've updated it